nixos/modules/services/networking/imaginary.nix

```diff
@@ -0,0 +1,110 @@
+{ lib, config, pkgs, utils, ... }:
+
+let
+  inherit (lib) mdDoc mkEnableOption mkIf mkOption types;
+
+  cfg = config.services.imaginary;
+in {
+  options.services.imaginary = {
+    enable = mkEnableOption (mdDoc "imaginary image processing microservice");
+
+    address = mkOption {
+      type = types.str;
+      default = "";
```

Comment:
Does emptry string equal to localhost? If I understand the nextcloud doc correct it sounds like you want to run imaginary always bound on localhost.

Reply:
Yes, it does mean that according to https://pkg.go.dev/net#Dial. I went with upstream's default: https://github.com/h2non/imaginary/blob/master/imaginary.go#L20

Reply:
Never mind, it means any address. That's also what it says in https://github.com/h2non/imaginary#command-line-usage.

Reply:
I guess it should be changed to `localhost` by default. See https://github.com/NixOS/nixpkgs/issues/100192. But the problem is that `localhost` means `127.0.0.1` and doesn't include `::1`.

Reply:
Can we use ``[::1]``?

Reply:
We can but what's the argument for one rather than the other?

Reply:
There was a magic ipv6 address which makes programs listen on both ipv4 and ipv6 address. I think it was ::1 or just ::.

Reply:
> `localhost` usually includes `::1` as well, at least if the application doesn't toss all but the first result of `getaddrinfo`

Doesn't work in this case:
```
$ curl 127.0.0.1:8088
{"imaginary":"1.2.4","bimg":"1.1.4","libvips":"8.14.1"}
$ curl [::1]:8088
curl: (7) Failed to connect to ::1 port 8088 after 0 ms: Couldn't connect to server
```

Reply:
Should we set `address = "localhost"` by default anyway?

Reply:
https://github.com/NixOS/nixpkgs/pull/217339

Reply:
For the nextcloud use case it is only accessed through localhost.